File: backend/app/models/SemanticEvaluation.py
from pydantic import BaseModel


class SemanticEvaluation(BaseModel):
    relevanceScore: float
    relevanceLabel: str
    categoryChoice: str
    directAnswerProbability: float
    preservesAuditProbability: float
    confidence: float
    combinedScore: float
    probabilityDistribution: dict[str, float]

File: frontend/src/types/SemanticEvaluation.ts
import { ProbabilityDistribution } from "@/types/ProbabilityDistribution"

export interface SemanticEvaluation {
  relevanceScore: number
  relevanceLabel: string
  categoryChoice: string
  directAnswerProbability: number
  preservesAuditProbability: number
  confidence: number
  combinedScore: number
  probabilityDistribution: ProbabilityDistribution
}
